NCS.DSS.Customer/AzureSearchDataSyncTrigger/CustomerSearchDataSyncTrigger.cs

```diff
@@ -1,91 +1,84 @@
+                        var customer = JsonSerializer.Deserialize<Models.CustomerSearch>(custJson);
+                        if(doc.CustomerId == null && doc.id != null)
+                            customer.CustomerId = doc.id;
+                        else if(doc.id == null && doc.CustomerId != null)
+                            customer.CustomerId = doc.CustomerId;
+
+                        customers.Add(customer);
+                    }
+                    var custFiltered = customers.Where(d => d.CustomerId != null);
+                    if (custFiltered.Any())
+                    {
 
+                        _logger.LogInformation("Attempting to Merge / Upload documents with IDs ({Ids}) to azure search", string.Join(',', custFiltered.Select(d => d.CustomerId).ToArray()));
+                        var batch = IndexDocumentsBatch.MergeOrUpload(custFiltered);
 
-                try
-                {
-                    _log.LogInformation("attempting to merge docs to azure search");
+                        _logger.LogInformation("Attempting to Index documents to azure search");
+                        var results = await client.IndexDocumentsAsync(batch);
 
-                    var results = await client.IndexDocumentsAsync(batch);
+                        var failed = results.Value.Results.Where(r => !r.Succeeded).Select(r => r.Key).ToList();
 
-                    var failed = results.Value.Results.Where(r => !r.Succeeded).Select(r => r.Key).ToList();
+                        if (failed.Count > 0)
+                        {
+                            _logger.LogInformation("Failed to Index some of the documents: {errors}", string.Join(", ", failed));
+                        }
 
-                    if (failed.Count > 0)
+                        _logger.LogInformation("Successfully Merged and Indexed documnets to azure search");
+                    }
+                    var custFailed = customers.Where(d => d.CustomerId == null);
+                    if (custFailed.Any())
                     {
-                        _loggerHelper.LogInformationMessage(_log, correlationId, string.Format("Failed to index some of the documents: {0}", string.Join(", ", failed)));
+                        foreach (var doc in custFailed.Where(d => d.CustomerId == null))
+                        {
+                            _logger.LogWarning("{Doc} missing Document Key (CustomerId) ", JsonSerializer.Serialize(doc));
+                        }
                     }
-
-                    _log.LogInformation("successfully merged docs to azure search");
-
                 }
-                catch (RequestFailedException e)
+                catch (Exception e)
                 {
-                    _loggerHelper.LogException(_log, correlationId, e);
-
-                }
-            }
+                    _logger.LogError("Request failed Excpetion with {error} {stacktrace}", e.Message, e.StackTrace);
```

Comment:
The stack trace is not required and the log should pass the exception object:
_logger.LogError(e, "Request failed. Exception with {error}", e.Message);